src/cli/list.rs

```diff
@@ -0,0 +1,283 @@
+//         console::style("Build").bold(),
+//         console::style("Size").bold(),
+//         console::style("Kind").bold(),
+//         console::style("Channel").bold(),
+//     );
+
+//     for package in packages {
+//         // Convert size to human readable format
+//         let size_human = match package.size_bytes {
+//             Some(size_bytes) => human_bytes(size_bytes as f64).to_string(),
+//             None => "".to_string(),
+//         };
+
+//         println!(
+//             "{:40} {:14} {:19} {:12} {:6} {:19}",
+//             if package.is_explicit {
+//                 console::style(package.name).green().bright().bold()
+//             } else {
+//                 console::style(package.name)
+//             },
+//             console::style(package.version),
+//             console::style(package.build.unwrap_or_else(|| "".to_string())),
+//             console::style(size_human),
+//             console::style(package.kind),
+//             console::style(package.channel.unwrap_or_else(|| "".to_string())),
+//         );
+//     }
+// }
+
+fn print_packages_as_table(packages: Vec<PackageToOutput>) {
+    // Initialize table
+    let mut table = Table::new();
+
+    table
+        .load_preset(NOTHING)
+        // .apply_modifier(UTF8_NO_BORDERS)
+        .set_content_arrangement(ContentArrangement::Dynamic);
+
+    // Add headers
+    table.set_header(vec![
+        Cell::new("Package").add_attribute(Attribute::Bold),
+        Cell::new("Version").add_attribute(Attribute::Bold),
+        Cell::new("Build").add_attribute(Attribute::Bold),
+        Cell::new("Size").add_attribute(Attribute::Bold),
+        Cell::new("Kind").add_attribute(Attribute::Bold),
+        Cell::new("Channel").add_attribute(Attribute::Bold),
+    ]);
+
+    for package in packages {
+        // Convert size to human readable format
+        let size_human = match package.size_bytes {
+            Some(size_bytes) => human_bytes(size_bytes as f64).to_string(),
+            None => "".to_string(),
+        };
+
+        let package_name = if package.is_explicit {
+            Cell::new(package.name)
+                .fg(Color::Green)
+                .add_attribute(Attribute::Bold)
+        } else {
+            Cell::new(package.name)
+        };
+
+        table.add_row(vec![
+            package_name,
+            Cell::new(package.version),
+            Cell::new(package.build.unwrap_or_else(|| "".to_string())),
+            Cell::new(size_human),
+            Cell::new(package.kind),
+            Cell::new(package.channel.unwrap_or_else(|| "".to_string())),
+        ]);
+    }
+
+    println!("{table}");
+}
+
+fn json_packages(packages: Vec<PackageToOutput>, json_pretty: bool) {
+    let json_string = if json_pretty {
+        serde_json::to_string_pretty(&packages)
+    } else {
+        serde_json::to_string(&packages)
+    }
+    .map_err(|_| miette::miette!("Cannot serialize packages to JSON"))
+    .unwrap();
```

Comment:
Maybe we can get rid of this unwrap :)?

Reply:
We prefer to bubble to errors up so we can pretty print them instead of the less user friendly assertions. Although I would think that failing to deserialize would be really unexpected so asserting would be fine here.

Nonetheless instead of `map_err` and `unwrap` an `expect` would be better I think.

```
    .expect("failed to serialize packages to json")
```

Reply:
Ok thanks!